app/assets/javascripts/govuk_publishing_components/components/intervention.js

```diff
@@ -0,0 +1,28 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function Intervention ($module) {
+    this.$module = $module
+    this.$banner = this.$module.querySelector('.gem-c-intervention')
+    this.$closeLink = this.$module.querySelector('.gem-c-intervention__dismiss-link')
+  }
+
+  Intervention.prototype.init = function () {
+    this.$module.close = this.handleClose.bind(this)
+
+    if (this.$closeLink) {
+      this.$closeLink.addEventListener('click', this.$module.close)
+    }
+  }
+
+  Intervention.prototype.handleClose = function (event) {
+    if (event) {
+      event.preventDefault()
+    }
+
+    this.$module.style.display = 'none'
```

Comment:
How does this not reappear if the page is reloaded? Or what's the plan for how that should work?

Reply:
Argh...

Reply:
(sorry)

Reply:
I can only think to set a cookie (if accepted)... I thought about also setting a query param but that wouldn't help on page refresh nor going to another page that also has the banner. Cookies it is then...

Reply:
https://github.com/alphagov/govuk_publishing_components/blob/main/app/assets/javascripts/govuk_publishing_components/lib/cookie-functions.js#L169 some potentially useful code here to help with that.

Reply:
oh _now_ I realise why doing this per app _might_ be better because you can be running multiple campaigns using the intervention component across GOV.UK so setting a cookie to remember if you dismissed one, if I go ahead and implement it the way I can only think here, would hide _all_ of them including ones the user _won't have seen yet_... Thoughts @andysellick ?

Reply:
One thing I could do I suppose is set the cookie name from a data attribute based on one of the options passed to the component?

Reply:
Yes, that could work. The JS would need to read that in and check the cookie to determine whether to allow the banner to be shown or not on page load. Not too complex, hopefully.